src/praisonai/praisonai/tool_resolver.py

```
@@ -569,7 +569,36 @@ def list_available(self) -> Dict[str, str]:
                         available[name] = "External tool from praisonai-tools"
             except (ImportError, AttributeError):
                 pass
-        
+
+        # 4. Add tools registered in the wrapper ToolRegistry (register_function API)
+        if self._registry is not None:
+            try:
+                names = self._registry.list_functions()
+            except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)